resources/service-info/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use shuttle_service::{error::Error, Factory, ResourceBuilder, Type};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ServiceInfo {
+    /// The Shuttle service name.
+    name: String,
```

Comment:
Could be called `service_name`, since we will want to extend this resource to include things like `project_name`, `deployment_id`, `public_url`, etc.